### PR TITLE
Template system

### PR DESCRIPTION
**Pull request by NikBelyaev**

### CODE REVIEW

File: src/blog_vi/templates/blog.html [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                                    by {{ blog.author_name }} on {{ blog.publish_date }} in
+
+                                    {% for category in blog.categories %}
+                                        <!-- TODO: category item styling -->
+                                        {{ category }}
+                                    {% endfor %}
+                                </p>
+                            </div>
+                        </div>
+                    </div>
```
Comment (NikBelyaev) [inline src/blog_vi/templates/blog.html:134]: Did you forget to remove this? This looks the same as the content of `card.html`

File: src/blog_vi/utils.py [hunk 2]
```diff
@@ -70,18 +71,34 @@ def get_md_file(text: str, file_name: str) -> str:
...
+        for directory in list(os.walk(app_templates_dir))[0][1]:
+            if directory == 'assets':
+                if directory in list(os.walk(templates_dir))[0][1]:
+                    break
+                shutil.copytree(app_templates_dir/directory, templates_dir/directory)
+
+        for file in app_files:
+            if file in template_files:
+                continue
+            shutil.copy(app_templates_dir/file, templates_dir)
```
Comment (NikBelyaev) [inline src/blog_vi/utils.py:102]: This seems really complicated. My suggestion is to use the [shutil.copytree()](https://docs.python.org/3/library/shutil.html#shutil.copytree) function with the custom `copy_function` defined. `copytree` accepts this keyword in order to change the copy behaviour (you can read about this in its documentation). So you need to write a custom function, that checks whether the file is already exists and do not copy, if yes.
Check out this [implementation (section 3.3)](https://bswen.com/2021/03/python-how-to-copy-file-without-overwrite-using-python-shutils.html), you can use it, but note, that the function accepts only `src` and `dst` args (also mentioned in the docs).

File: src/blog_vi/utils.py [hunk 3]
```diff
@@ -69,19 +70,31 @@ def get_md_file(text: str, file_name: str) -> str:
...
     workdir.joinpath('articles').mkdir(exist_ok=True)
 
     templates_dir = workdir / 'templates'
+    app_templates_dir = Path(__file__).parent / 'templates'
     if not templates_dir.exists():
-        app_templates_dir = Path(__file__).parent / 'templates'
 
         # Move the default template dir to the current workdir
         shutil.copytree(app_templates_dir, templates_dir)
+    else:
```
Comment (NikBelyaev) [inline src/blog_vi/utils.py:95]: You can get rid off this conditional, just `shutil.copytree(app_templates_dir, templates_dir, copy_function=copy_, dirs_exist_ok=True)`

File: src/blog_vi/utils.py [hunk 4]
```diff
@@ -69,19 +70,25 @@ def get_md_file(text: str, file_name: str) -> str:
     return file_name
 
 
+def copy_(src, dst):
```
Comment (NikBelyaev) [inline src/blog_vi/utils.py:73]: rename it to `copy_without_overwrite`

File: src/blog_vi/templates/article.html [hunk 5]
```diff
@@ -29,12 +30,12 @@
   <meta property="twitter:image" content="{{ head_blog.header_image }}">
 
   <link
-    href="../../templates/css/tailwind.min.css"
+    href="templates/assets/css/tailwind.min.css"
```
Comment (NikBelyaev) [inline src/blog_vi/templates/article.html:33]: This won't work as long as we store article htmls in the different directory from the root. So make this path relative as before.

File: src/blog_vi/templates/article.html [hunk 6]
```diff
@@ -29,12 +30,12 @@
...
     rel="stylesheet"
   />
   <link
     rel="stylesheet"
-    href="../../templates/css/typography.min.css"
+    href="templates/assets/css/typography.min.css"
```
Comment (NikBelyaev) [inline src/blog_vi/templates/article.html:38]: The same as above

**Review by PhilippeTrounev — APPROVED** [review]

**Pull request merged**